src/getIdentifier.test.ts

```diff
@@ -0,0 +1,151 @@
+import { v3 } from "@govtechsg/open-attestation";
+import { getIdentityProofFragment, getIdentifier } from "./getIdentifier";
+import { VerificationFragment } from "./types/core";
+import {
+  openAttestationDnsTxtIdentityProof,
+  openAttestationDnsDidIdentityProof,
+  openAttestationDidIdentityProof,
+} from ".";
+import { getProvider } from "./common/utils";
+
+import v3DidSignedRaw from "../test/fixtures/v3/did-signed.json";
+import v3DnsDidSignedRaw from "../test/fixtures/v3/dnsdid-signed.json";
+import v3DocumentStoreIssuedRaw from "../test/fixtures/v3/documentStore-issued.json";
+
+const v3DidSigned = v3DidSignedRaw as v3.SignedWrappedDocument;
+const v3DnsDidSigned = v3DnsDidSignedRaw as v3.SignedWrappedDocument;
+const v3DocumentStoreIssued = v3DocumentStoreIssuedRaw as v3.WrappedDocument;
+
+const options = {
+  provider: getProvider({ network: "ropsten" }),
+};
+
+const verificationFragment1: VerificationFragment = {
+  status: "SKIPPED",
+  type: "DOCUMENT_STATUS",
+  name: "OpenAttestationDidSignedDocumentStatus",
+  reason: {
+    code: 0,
+    codeString: "SKIPPED",
+    message: "Document was not signed by DID directly",
+  },
+};
+const verificationFragment2: VerificationFragment = {
+  name: "OpenAttestationDnsTxtIdentityProof",
+  type: "ISSUER_IDENTITY",
+  data: [
+    {
+      status: "VALID",
+      location: "example.openattestation.com",
+      value: "0x2f60375e8144e16Adf1979936301D8341D58C36C",
+    },
+  ],
+  status: "VALID",
+};
+const verificationFragment3: VerificationFragment = {
+  name: "OpenAttestationDnsDidIdentityProof",
+  type: "ISSUER_IDENTITY",
+  data: [
+    {
+      location: "example.tradetrust.io",
+      key: "did:ethr:0xE712878f6E8d5d4F9e87E10DA604F9cB564C9a89#controller",
+      status: "VALID",
+    },
+  ],
+  status: "VALID",
+};
+const verificationFragment4: VerificationFragment = {
+  name: "OpenAttestationDidIdentityProof",
+  type: "ISSUER_IDENTITY",
+  data: [
+    {
+      did: "did:ethr:0xE712878f6E8d5d4F9e87E10DA604F9cB564C9a89",
+      status: "VALID",
+    },
+  ],
+  status: "VALID",
+};
+const malformedVerificationFragment1: VerificationFragment = {
```

Comment:
let's add a comment to explain why this fragment is malformed, same for the other

Reply:
sure